src/components/FacetInsights.js

```diff
@@ -124,14 +126,32 @@ export default class FacetInsights extends React.Component<FacetInsightsDefaultP
     return 'list';
   }
 
-  facetForField(fieldName: string, facetMap: Map<string, SearchFacet>) {
+  /**
+   * Filter Buckets in Facet that are already applied to the search filters.
+   * Buckets of type tagcloud and timeseries do not need to be filtered.
+   */
+  filterFacetBuckets = (facet: SearchFacet, type: string, facetFiltersMap: Map<string, FacetFilter>): SearchFacet => {
+    if (type === 'tagcloud' || type === 'timeseries') {
+      return facet;
+    }
+    const filteredBuckets = [];
+    facet.buckets.forEach((bucket: SearchFacetBucket) => {
+      if (!facetFiltersMap.get(bucket.filter)) {
+        filteredBuckets.push(bucket);
+      }
+    });
+    return Object.assign(facet, { buckets: filteredBuckets });
```

Comment:
Instead of `Object.assign`
```suggestion
    return { 
      ...facet, // spread operator
      buckets: filteredBuckets, // order matters - This will overwrite facet.buckets
    });
```

Reply:
Not sure why, but flow seems to complain about this syntax.

```
return {
142: ...facet,
143: buckets: filteredBuckets,
144: };
^ object literal. This type is incompatible with the expected return type of
132: filterFacetBuckets = (facet: SearchFacet, type: string, facetFiltersMap: Map<string, FacetFilter>): SearchFacet => {
^^^^^^^^^^^ SearchFacet
```

Reply:
My example is returning an object: `{}`, while the flow declaration expects type `SearchFacet`.

Reply:
You'll need to either return a `SearchFacet` instead of an object or update the return type in the flow declaration.

Reply:
Addressed this by returning `new SearchFacet()`